test/e2e/infinispan/upgrade_operand_test.go

```diff
@@ -407,6 +408,50 @@ func TestSpecImageUpdate(t *testing.T) {
 	assert.EqualValues(t, 1, ss.Status.ObservedGeneration)
 }
 
+// TestPodAlreadyShutdownOnUpgrade simulates a scenario where a GracefulShutdown fails when only a subset of pods have had
+// their container shutdown
+func TestPodAlreadyShutdownOnUpgrade(t *testing.T) {
+	defer testKube.CleanNamespaceAndLogOnPanic(t, tutils.Namespace)
+
+	i := tutils.DefaultSpec(t, testKube, func(infinispan *ispnv1.Infinispan) {
+		infinispan.Spec.Replicas = 1
+		infinispan.Spec.Security.EndpointAuthentication = pointer.Bool(false)
+	})
+	testKube.CreateInfinispan(i, tutils.Namespace)
+	testKube.WaitForInfinispanPods(1, tutils.SinglePodTimeout, i.Name, tutils.Namespace)
+	i = testKube.WaitForInfinispanCondition(i.Name, i.Namespace, ispnv1.ConditionWellFormed)
+
+	schema := i.GetEndpointScheme()
+	client_ := testKube.WaitForExternalService(i, tutils.RouteTimeout, tutils.NewHTTPClientNoAuth(schema), nil)
+	ispnClient := v15.New(client_)
```

Comment:
Any reasonable way to use the client based on the version being tested to future proof the test? And ideally disable the test if it's not supported (but I don't think that's the case with Infinispan 14 and we don't support older operands)

Reply:
Yes, good catch. We can do `ispnClient := client.New(tutils.CurrentOperand, client_)` and that will use the currently configured Operand to create the client.